### PR TITLE
feat: adds Taraxa staking and tvl adapter

### PR DESCRIPTION
**Pull request by rattrap**
##### Name (to be shown on DefiLlama):

Taraxa

##### Twitter Link:

https://x.com/taraxa_project

##### List of audit links if any:

https://www.halborn.com/audits/taraxa/node-and-evm-l1-security-assessment

##### Website Link:

https://taraxa.io/

##### Logo (High resolution, will be shown with rounded borders):

https://drive.google.com/file/d/1dnhJ9rdzFT9PRYp0r7wXkg3nIJNpIl0M/view?usp=sharing

##### Current TVL:

7182603238 TARA ($69.937.000)

##### Treasury Addresses (if the protocol has treasury)

0x10f4fd4d9856efd5700f0cb70b90bf519a3cd238

##### Chain:

Taraxa (841)

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): 

taraxa

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): 

8715

##### Short Description (to be shown on DefiLlama):

PoS, blockDAG, EVM-compatible Layer-1

##### Token address and ticker if any:

-

##### Category (full list at https://defillama.com/categories) *Please choose only one:

[Chain](https://defillama.com/protocols/Chain)

##### Github org/user (Optional, if your code is open source, we can track activity):

Taraxa-project

### CODE REVIEW

File: projects/taraxa/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const balance = await api.provider.getBalance(
+    TARAXA_STAKING_CONTRACT,
+    api.block
+  );
+  api.addGasToken(Number(balance) / 10 ** 18);
+}
+
+async function tvl(api) {
+  const balanceEcosystem = await api.provider.getBalance(
+    TARAXA_ECOSYSTEM_FUND_ADDRESS,
```
Comment (g1nt0ki) [inline projects/taraxa/index.js:16]: is this your treasury address?

File: projects/taraxa/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const TARAXA_STAKING_CONTRACT = "0x00000000000000000000000000000000000000fe";
+const TARAXA_ECOSYSTEM_FUND_ADDRESS =
+  "0x723304d1357a2334fcf902aa3d232f5139080a1b";
+const TARAXA_FOUNDATION_ADDRESS = "0x723304d1357a2334fcf902aa3d232f5139080a1b";
+
+async function staking(api) {
```
Comment (g1nt0ki) [inline projects/taraxa/index.js:6]: sorry, we are not tracking chain staking atm

**Comment by llamatester** [comment]
The adapter at projects/taraxa exports TVL: 
        
 
 
    tara-staking              0
    tara                      0
    staking                   0
    
    total                    0

File: projects/taraxa/index.js (hunk 1 above)
Comment (rattrap) [reply projects/taraxa/index.js:16]: The "Foundation" address is the treasury. This is a community fund.

**Comment by rattrap** [comment]
@g1nt0ki, have any ideea why it's returning 0 TVL?

File: projects/taraxa/index.js (hunk 1 above)
Comment (rattrap) [reply projects/taraxa/index.js:16]: Actually both of them count as the project treasury. The foundation tokens are never circulated and the ecosystem fund is for grants (this changes from month to month)

Comment (g1nt0ki) [reply projects/taraxa/index.js:16]: okay, we can track this under treasury but not as tvl. making the change now

Comment (g1nt0ki) [reply projects/taraxa/index.js:16]: cant count foundation tokens as it is not circulating

**Comment by g1nt0ki** [comment]
fixed it here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/a52c1fb6352804bb5d6e57864a06c20402d6ba29

**Pull request closed without merging**